app/perfunwind.cpp

```diff
@@ -682,6 +693,27 @@ void PerfUnwind::analyze(const PerfRecordSample &sample)
     sendBuffer(buffer);
 }
 
+void PerfUnwind::updateIpCache()
+{
+    const auto& stack = m_currentUnwind.ipStack;
+    for (int i = 0; i < stack.length() - 1; i++) {
+        m_ipCache[m_currentUnwind.sample->pid()][stack[i].address] = {m_currentUnwind.frames, stack[i].frameOffset};
+    }
+}
+
+void PerfUnwind::stitchCurrentCallchain()
+{
+    if (m_currentUnwind.ipStack.isEmpty())
+        return;
+
+    const auto& cache = m_ipCache[m_currentUnwind.sample->pid()];
+    auto it = cache.find(m_currentUnwind.ipStack.last().address);
+
+    if (it != cache.end()) {
+        it->appendFramesTo(m_currentUnwind.frames);
```

Comment:
this will also need to update the ipStack for the current unwind, as otherwise updateIpCache will not work correctly, no?

Reply:
ipStack does not need to be updated (I think by updated you mean to add the ips of the stitched samples). If the current call chain can be stitched, the new frames are added to current unwind. Then when updateIpCache is called, the addresses from the current sample along with the longest known extended call chain are stored in the cache. If ipStack would be updated, it would overwrite values in ipCache without adding new information.

Reply:
I see, please add that info to a comment in the code